build.gradle.kts

```diff
@@ -18,197 +14,296 @@ import org.gradle.api.tasks.testing.logging.TestExceptionFormat.FULL
  * limitations under the License.
  *
  */
+import net.researchgate.release.GitAdapter.GitConfig
+import java.util.Date
+import org.gradle.api.tasks.testing.logging.TestExceptionFormat.FULL
 
 plugins {
-  id("java")
-  id("maven")
-  id("java-gradle-plugin")
-  id("net.researchgate.release") version "2.6.0"
-  id("com.github.sherter.google-java-format") version "0.8"
-  id("checkstyle")
-  id("jacoco")
+    id("java")
```

Comment:
Can you please point me to Google style guide for Kotlin that says "2 spaces"?

https://github.com/google/styleguide/issues/314 says Google style is (see https://github.com/google/styleguide/pull/365/files) https://developer.android.com/kotlin/style-guide, which says "4 spaces" https://developer.android.com/kotlin/style-guide#indentation just like the Kotlin official coding conventions: https://kotlinlang.org/docs/coding-conventions.html#indentation

Reply:
@elefeint , would you please clarify if "squashing on merge" is an absolute requirement?
Squashing commits breaks git history analysis, and committing individual changes as separate commits helps that.

Do you suggest I should file 8 individual PRs?

The squash in https://github.com/GoogleCloudPlatform/app-gradle-plugin/pull/423#issuecomment-1242909704 caused `build.gradle.kts` to look like "a completely fresh file written by @martinbonnin" even though Martin tried their best to translate to Kotlin one-to-one.

----

> More importantly, keeping unrelated style changes out of PRs makes the inevitable code archaeology so much easier

In fact, IntelliJ IDEA shows annotate for the "re-indented" lines just fine:

<img width="798" alt="git annotate in IDEA for build.gradle.kts" src="https://user-images.githubusercontent.com/213894/189517629-e2c4287f-86d0-40e3-b499-3c35ca9d06d0.png">

As you see, the indentation is new, however, "commit line" attribution is proper (it does not show all the lines as if they were modified by me.

GitHub UI is slightly different, and it does treat re-indent as commit that changes the line: https://github.com/vlsi/app-gradle-plugin/blame/refactor_publication/build.gradle.kts#L61, however, if the changes are NOT squashed, then there's a convenient "show changes prior the commit" button that shows the annotations prior to re-indent: https://github.com/vlsi/app-gradle-plugin/blame/2a49f56b03fa5c71343f829efd189debb3b9bcc3/build.gradle.kts

I incline that committing individual changes as separate commits helps for `code archaeology`, and re-indenting as a separate commit does not make it much harder.

Reply:
Squashing is not an absolute requirement; merely a common org/project practice. Since your individual commits are well structured, I can rebase-merge this pull request specifically.
For the indentation, though, keeping 2 space indents would be ideal. We will work on externalizing Google's Kotlin style guide with @jamesward; in the meantime you'll have to take my word for it, I'm afraid.